base/socket.jl

```diff
@@ -582,13 +582,27 @@ function getaddrinfo(cb::Function, host::ASCIIString)
 end
 getaddrinfo(cb::Function, host::AbstractString) = getaddrinfo(cb,ascii(host))
 
+
+type DNSError <: Exception
+    uverror::UVError
+    host::AbstractString
+end
+
+function show(io::IO, err::DNSError)
+    print(io, "DNSError: ", err.host, ", ")
+    show(io, err.uverror)
+end
+
 function getaddrinfo(host::ASCIIString)
     c = Condition()
     getaddrinfo(host) do IP
         notify(c,IP)
     end
     ip = wait(c)
-    isa(ip,UVError) && throw(ip)
+    if isa(ip,UVError)
+        @assert ip.code in [UV_EAI_NONAME, UV_EAI_AGAIN, UV_EAI_FAIL, UV_EAI_NODATA]
```

Comment:
Again, an assert is not appropriate here. If the code is not in this list, rethrow it so the user can see what the code actually was.